testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionInitializerTest.java

```diff
@@ -210,13 +218,29 @@ public void testUserSessionInitializerWithDeletingClient(KeycloakSession session
             KeycloakSession currentSession = inheritClientConnection(session, createSessionPersister3);
             RealmModel realm = currentSession.realms().getRealmByName(realmName);
 
-            // Delete cache (persisted sessions are still kept)
-            currentSession.sessions().onRealmRemoved(realm);
+            // Delete user/client[offline] cache (persisted sessions are still kept)
+            InfinispanConnectionProvider provider = currentSession.getProvider(InfinispanConnectionProvider.class);
```

Comment:
I am not sure if it is better to keep all the logic in the test or rather make sure that test is able to call InfinispanUserSessionProvider.removeLocalUserSessions, which is private method? WDYT @hmlnarik . My vote is to rather call the InfinispanUserSessionProvider to remove some logic from the test. Maybe we can just make that method on InfinispanUserSessionProvider public with the comment to have it public for testing purposes only? Another option is reflection, but that is not so ideal as well...

Reply:
Yup, I agree. That sounds as a better option, thanks for spotting this!

Reply:
Thank you Marek for the review. I implemented both your suggestions.